rfcs/text/remove-support-for-uncontrolled-inputs.md

```diff
@@ -0,0 +1,234 @@
+};
+
+const MyComponent = ({ value, onChange, ...rest }: MyComponentProps) => {
+   const handleOnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    if (onChange) {
+      onChange(event.target.value);
+    } 
+  };
+
+  /* Component logic goes here */
+
+  return <input value={value} onChange={handleOnChange} />;
+};
+
+export default MyComponent;
+```
+# Detailed design
+
+- We will add deprecation warnings to our input components so that our consumers are aware of the pending changes to remove uncontrolled support. This will appear as a console warning if our consumer is using the uncontrolled mode and a warning on our Storybook documentation.
+
+- We will remove uncontrolled behaviour from our input components and associated props. This will be done in one Pull Request with each component having it's own commit message with a `BREAKING CHANGE`. The benefit of this is that we will only release one major version of Carbon. 
+
+This approach will provide immediate benefit to developers of Carbon as it means we will have less code and tests to maintain. It will also mean that consumers of Carbon can steadily upgrade their version of Carbon.
+
+## Components
+
+The input components that currently support being uncontrolled are: 
+1. Textbox
+1. Decimal
+1. Number Input
+1. Simple Select
+1. Filterable Select
+1. Multi Select
+1. Grouped Character
+1. Numeral Date
+1. Search
+1. Checkbox 
+1. RadioButton 
+1. Switch 
+1. SimpleColorPicker 
+1. AdvancedColorPicker 
+1. ButtonToggle
+
+## Unit Testing
+
+Currently we have unit tests for both controlled and uncontrolled behaviours. As part of this work we could reduce the amount of code in the spec files quite considerably. If we take `Decimal` as an example of this, `Decimal` currently has 1430 lines of code in the spec file. This could be reduced to down to around 480 lines of code if we were to remove the uncontrolled tests. This would make the tests easier to read and maintain, while also reducing the runtime of the tests.
+# Drawbacks
+
+- This work will be a `BREAKING CHANGE`, which we will communicate in the appropriate Carbon channels. Consumers of Carbon will also have the opportunity to comment on this RFC.
+
+- Any consumers of Carbon that currently use uncontrolled components will have to rewrite their form implementations.  
+
+# Alternatives
+
+The alternative is to do nothing: we leave the input components as they are and continue to support both controlled and uncontrolled behaviours. This course of action would mean no impact to our consumers. 
+
+Uncontrolled input components can be used with Formik, even though this is not encouraged, through implementing a store type (React's [useReducer hook](https://reactjs.org/docs/hooks-reference.html#usereducer) or [Redux](https://redux.js.org/) for example) mechanism in order to retrieve a value. However, we would always still recommend using controlled input components with Formik.
```

Comment:
I don't understand this paragraph. To use a reducer you'd need to use a controlled input, unless you're meaning to use `onChange` to maintain the reducer state?

Reply:
Removed this paragraph as discussed.